src/directions/main.ts

```diff
@@ -1,20 +1,18 @@
-import type maplibregl from "maplibre-gl";
-import type { Directions, MapLibreGlDirectionsConfiguration, Route, Snappoint } from "./types";
+import type { Map, MapGeoJSONFeature, MapMouseEvent, MapTouchEvent } from "maplibre-gl";
+import type { Directions, MapLibreGlDirectionsConfiguration } from "./types";
```

Comment:
So this basically allows tree-shaking?

Reply:
The `type` imports only import type information (typescript-related), which is obviously doesn't make its way into the final bundle (because the final bundle is JS and all the TypeScript-related stuff is purely for the build-time, not runtime).

Reply:
The original issue with tree-shaking the `Evented` super-class from maplibre-gl dep. is not resolved anyhow. I don't know whether it's MapLibre's issue, or the issue of build-tooling I'm using. Instead of taking `Evented` from maplibre, I implemented my own `Evented`.

Reply:
I think there is no tree shaking in MapLibre GL JS. https://github.com/maplibre/maplibre-gl-js/issues/977

Reply:
Ah, yeah. That makes things clearer. Anyway, I didn't do any magic there and didn't choose a custom implementation for `Evented`, so if there at any point a version of GL JS that supports tree-shaking, I'll be happy to simply use that exported super class instead of supporting my own implementation. But at the time being I see my solution as the only option. Loosing more than 600kB in bundle size because of the only class is not a good thing for sure.